website/events/models/event_registration.py

```diff
@@ -110,17 +115,26 @@ def information_fields(self):
     def is_registered(self):
         return self.date_cancelled is None
 
-    @property
-    def queue_position(self):
-        if self.event.max_participants is not None:
-            try:
-                queue_ids = [
-                    registration.member_id for registration in self.event.queue
-                ]
-                return list(queue_ids).index(self.member_id) + 1
-            except ValueError:
-                pass
-        return None
+    queue_position = AnnotationProperty(
+        # Get queue position by counting amount of registrations with lower date and in case of same date lower id
+        # Subsequently cast to None if this is 0 or lower, in which case it isn't in the queue
+        NullIf(
+            Greatest(
+                Count(
+                    "event__eventregistration",
+                    filter=Q(event__eventregistration__date_cancelled=None)
+                    & Q(event__eventregistration__date__lt=F("date"))
+                    | (
+                        Q(event__eventregistration__id__lte=F("id"))
+                        & Q(event__eventregistration__date__exact=F("date"))
+                    ),
```

Comment:
```suggestion
                    filter=Q(event__eventregistration__date_cancelled=None)
                    & (Q(event__eventregistration__date__lt=F("date"))
                    | 
                        Q(event__eventregistration__id__lte=F("id"))
                        & Q(event__eventregistration__date__exact=F("date"))
                    ),
```
I believe `&` has precedence over `|`

Reply:
Mm maybe.  This is too complicated, I'll look at this Wednesday

Reply:
You currently have:

`( if the registration is not cancelled 'and' the registration date is before F("date") ) 'or' ( if the event registration has the same registration date 'and' the id is lower than F("id") )`

I suggested:

`if the registration is not cancelled 'and' ( the registration date is before F("date") 'or' ( the registration date is the same 'and' the id is smaller ) )`

Reply:
Yes you're right. Very keen eye haha
It would be a very weird bug if this were to cause confusion on production...